### PR TITLE
Remove (most) hardcoded dependencies on -[OBAApplication sharedApplication]

### LINKED ISSUE #742: Remove hardcoded dependencies on -[OBAApplication sharedApplication]
I've slowly been replacing hardcoded references to singleton objects like `[OBAApplication sharedApplication].modelService` with lazily loaded properties across the app. I've been doing this in order to make it easier to test our objects in isolation without the overhead or churn associated with going full-on Java-esque dependency injection.

Take a peek at `OBABookmarkGroupsViewController` for an example of how I implement this: The header file has a property for `OBAModelDAO`, and the implementation file has a lazily loading accessor. If someone is inclined, they can set `modelDAO` on the newly created bookmark groups controller before it is first accessed. Otherwise, the first time that the `modelDAO` property is accessed, the `[OBAApplication sharedApplication].modelDAO` singleton will be strongly referenced and returned.

The following is a list of every file where a property on `-[OBAApplication sharedApplication]` is accessed. There will absolutely be some false positives on this, just FYI.

- [x] OBAApplicationDelegate.m
- [x] OBAKit/Location/OBARegionHelper.m
- [x] ui/bookmark_groups/OBABookmarkGroupsViewController.m
- [x] ui/bookmark_groups/OBAEditBookmarkGroupViewController.m
- [x] ui/bookmark_groups/OBAEditStopBookmarkGroupViewController.m
- [x] ui/bookmarks/OBABookmarkRouteDisambiguationViewController.m
- [x] ui/bookmarks/OBABookmarksViewController.m
- [x] ui/bookmarks/OBAEditStopBookmarkViewController.m
- [x] ui/info/OBAAgenciesListViewController.m
- [x] ui/info/OBAInfoViewController.m
- [x] ui/recent_stops/OBARecentStopsViewController.m
- [x] ui/search/OBASearchController.m
- [x] ui/search/OBASearchResultsMapViewController.m
- [x] ui/situations/OBADiversionViewController.m
- [x] ui/situations/OBASituationViewController.m
- [x] ui/stop_details/OBAEditStopPreferencesViewController.m
- [x] ui/stop_details/OBAReportProblemWithRecentTripsViewController.m
- [x] ui/stop_details/OBAReportProblemWithStopViewController.m
- [x] ui/stops/OBAStopViewController.m
- [x] ui/trip_details/OBAArrivalAndDepartureViewController.m
- [x] ui/trip_details/OBAReportProblemWithTripViewController.m
- [x] ui/trip_details/OBATripDetailsViewController.m
- [x] ui/trip_details/OBATripScheduleListViewController.m
- [x] ui/trip_details/OBATripScheduleMapViewController.m
- [x] ui/trip_details/OBAVehicleDetailsController.m

### PR DESCRIPTION
**Pull request by aaronbrethorst**
Fixes #742 - Remove hardcoded dependencies on -[OBAApplication sharedApplication]

I've replaced almost all of the hardcoded references to singleton objects like [OBAApplication sharedApplication].modelService with lazily loaded properties across the app. I've been doing this in order to make it easier to test our objects in isolation without the overhead or churn associated with going full-on Java-esque dependency injection.

### CODE REVIEW

File: ui/bookmark_groups/OBAEditStopBookmarkGroupViewController.h [hunk 1]
```diff
@@ -18,12 +18,12 @@ NS_ASSUME_NONNULL_BEGIN
 @end
 
 @interface OBAEditStopBookmarkGroupViewController : UITableViewController <UITextFieldDelegate>
-@property (nonatomic,strong) NSArray *groups;
-@property (nonatomic,strong,nullable) OBABookmarkGroup *selectedGroup;
-@property (nonatomic,strong) id <OBABookmarkGroupVCDelegate> delegate;
+@property(nonatomic,strong) OBAModelDAO *modelDAO;
```
Comment (chadsy) [inline ui/bookmark_groups/OBAEditStopBookmarkGroupViewController.h]: `readonly`

File: ui/bookmarks/OBAEditStopBookmarkViewController.h [hunk 2]
```diff
@@ -14,15 +14,15 @@
...
+#import <OBAKit/OBAKit.h>
 #import "OBAEditStopBookmarkGroupViewController.h"
 
-@class OBAArrivalAndDepartureV2;
-
 NS_ASSUME_NONNULL_BEGIN
 
 @interface OBAEditStopBookmarkViewController : UITableViewController <OBABookmarkGroupVCDelegate>
+@property(nonatomic,strong) OBAModelDAO *modelDAO;
+@property(nonatomic,strong) OBAModelService *modelService;
```
Comment (chadsy) [inline ui/bookmarks/OBAEditStopBookmarkViewController.h]: `readonly` for these, too.

**Comment by aaronbrethorst** [comment]
@chadsy I'm actually looking to make them world writable so another object could poke a model service, model DAO, or location manager. Traditionally those would become parameters on init but that's a lot of overhead for what will almost always be from the singleton. Happy to elaborate on explanation if you'd like.

**Comment by chadsy** [comment]
No worries. Ship it!

**Pull request merged**